packages/components/src/internal/renderers.tsx

```diff
@@ -98,169 +115,303 @@ export const EditableColumnTitle: FC<EditableColumnTitleProps> = memo(props => {
+    // HeaderCellDropdown, which will reset the open value to true, which will keep the menu open.
+    const openFilterPanel = useCallback(
+        (_: any, event: SyntheticEvent) => {
+            cancelEvent(event);
             setOpen(false);
-        };
-    }, []);
-
-    const onToggleClick = useCallback(
-        (shouldOpen: boolean, evt?: any) => {
-            if (!includeDropdown || editingTitle) return;
-
-            // when menu is closed skip any clicks on icons by just checking for span el type
-            if (shouldOpen && evt && evt.target.tagName.toLowerCase() !== 'span') return;
-
-            setOpen(shouldOpen);
+            handleFilter(queryColumn, false);
         },
-        [includeDropdown]
-    );
-
-    const _handleFilter = useCallback(
-        (remove?: boolean) => {
-            handleFilter(col, remove);
+        [queryColumn, handleFilter]
+    ) as SelectCallback;
+    const removeFilter = useCallback(
+        (_: any, event: SyntheticEvent) => {
+            cancelEvent(event);
             setOpen(false);
+            handleFilter(queryColumn, true);
         },
-        [col, handleFilter]
-    );
+        [queryColumn, handleFilter, setOpen]
+    ) as SelectCallback;
 
-    const _handleSort = useCallback(
-        (dir?: string) => {
-            handleSort(col, dir);
+    const sort = useCallback(
+        (event: SyntheticEvent, dir?: string) => {
+            cancelEvent(event);
             setOpen(false);
+            handleSort(queryColumn, dir);
         },
-        [col, handleSort]
+        [queryColumn, handleSort, setOpen]
     );
+    // There is something wrong with the React Bootstrap types, the only way to get these callbacks properly typed is to
+    // use "as SelectCallback", even though their type signature matches perfectly.
+    const sortAsc = useCallback((_: any, event: SyntheticEvent): void => sort(event, '+'), [sort]) as SelectCallback;
+    const sortDesc = useCallback((_: any, event: SyntheticEvent): void => sort(event, '-'), [sort]) as SelectCallback;
+    const clearSort = useCallback((_: any, event: SyntheticEvent): void => sort(event), [sort]) as SelectCallback;
+    const hideColumn = useCallback(
+        (_: any, event: SyntheticEvent): void => {
+            cancelEvent(event);
+            setOpen(false);
+            handleHideColumn(queryColumn);
+        },
+        [queryColumn, handleHideColumn, setOpen]
+    ) as SelectCallback;
+    const addColumn = useCallback(
+        (_: any, event: SyntheticEvent): void => {
+            cancelEvent(event);
+            setOpen(false);
+            handleAddColumn(queryColumn);
+        },
+        [queryColumn, handleAddColumn, setOpen]
+    ) as SelectCallback;
+    const editColumnTitle = useCallback(
+        (_: any, event: SyntheticEvent): void => {
+            cancelEvent(event);
+            onEditTitleClicked();
```

Comment:
no need to call `setOpen(false);` here?

Reply:
No, the `onEditTitleClicked` method passed to this component closes the menu.